### ISSUE TITLE
Plotting on meshes

### ISSUE THREAD

**Issue by rowanc1** [simpeg/discretize#15]
_From @lheagy on June 2, 2016 19:48_

@thast is visualizing inversion results on SimPEG meshes and has a couple requests
- [ ] `showPadding = False` we should be able to detect where cell sizes change (for tensor at least) and only plot the core region of the mesh (default can be true) - this is important when showing inversion results (there is no data support in the padding). Sometimes you do want to show larger axes to compare lines that are offset but geo-referenced
- [ ] `mesh.plotCountours` same as plotimage - but do contours (or countouf) instead 

_Copied from original issue: simpeg/simpeg#330_

**Comment by rowanc1** [comment]
_From @thast on June 2, 2016 20:6_

![npmask](https://cloud.githubusercontent.com/assets/13682747/15759238/d85134f6-28c2-11e6-858f-cb05b6c553e0.png)

**Comment by rowanc1** [comment]
_From @thast on June 2, 2016 20:7_

masking can be use to remove the padding (as long as you know where to cut):
mask0 = np.ma.masked_where(np.logical_or(meshlist[0].gridCC[:,0]<641000,meshlist[0].gridCC[:,0]>644000),np.random.rand(len(modelDClist_work[0])))

**Comment by sarahgarre** [comment]
Hi all, is this contour stuff implemented in the end or is it not available yet? I would like to plot DOI_index with contours using plotImage with a treemesh, since the location where the contour lines are getting close to each other is often used as cut-off...

**Comment by micmitch** [comment]
@sarahgarre I think that you should have the functionality to do this. The DOI_index vector is probably the same size as your model so plot it the same way you would a model and then add the contours. More work is still needed to be able to plot vector field sections on the tree mesh. If you have trouble getting the plot you want on the tree mesh it is pretty easy to interpolate onto a small core tensor mesh and then plot. I can send you code for this if that would be helpful. Not sure how well documented it is though... :)

**Comment by sarahgarre** [comment]
@micmitch  yes, indeed. however, 
```
out = meshList20[i].plotImage(doi, ax=ax2,grid=False, 
                     clim=(0, 1),  
                     gridOpts={'alpha':0.2},ncontour=5)
```
does not give me any contours...?

**Comment by micmitch** [comment]
@sarahgarre  I don't think that I have actually added contours to a model section before.  hmmm.... just looked through plotImage. It doesn't look like it has an option for plotting contours.  It should not be too difficult to add this functionality though. This could probably be done by adding a contourOptions keyword argument to plotImage and calling plt.contour(contourOptions) if contourOptions is not None.

Looks like we are using pcolormesh for the actual plotting. This link might be helpful for overlying contours.

https://stackoverflow.com/questions/34389363/plotting-contours-over-pcolormesh-data
